151 footer

sherakama opened the pull request:
# READY FOR REVIEW

# Summary
- Adds new Ligature Font
- Adds new ligature-logo mixin
- Adds new global footer organism (scss & twig)
- Adds new centered-grid-container mixin for centering an item with css grid
- Adds new grid-media mixin for fast creating grids at different breakpoints
- Adds new mixin section in styleguide called "Icons and Logos"
- Fixed up the twig conventions on `{{ attributes }}` and `{{ modifier_class }}`
- Updated the namespace to use the new core directory with twig includes
- Adds new logo twig template (atom)

The "sticky" part of this footer has been intentionally left out of this pull request.

# Steps to Test

1. Check out this branch
2. Fire up your web browser and have a look at the global footer in full page view (http://decanter.sws/item-organisms-global-footer.html)
3. Visually review for aesthetics. 
4. Review HTML & Twig markup for global-footer
5. Review SCSS markup for global-footer
6. Review the rest of the code in this PR and in particular pay attention to the new grid mixins
7. Make comments on my review comments

# Associated Issues and/or People
- @bixgomez 
- @josephgknox 
- @JBCSU 

# Notes
Some General thoughts and notes with working on this component:

1. Grid system is lacking helpers and requires too much code writing. I found it difficult to create a responsive footer element and ended up writing a couple of mixins to help. We should shore up some of the syntax so that we are writing less straight up code into our component and use common mixins.
2. Components could use some refactoring to tease out common elements and made into smaller molecules and atoms. We have a need for atoms and molecules so that we can put together organisms. 
3. There are differing conventions on the twig templates for attributes and classes. Every block level element should have an `{{ attributes }}` twig variable and include a `{{ modifier_class }}` twig variable in the class list. I have remedied this for now but we should document it somewhere.

sherakama commented on core/scss/components/organisms/_global-footer.scss:
```diff
@@ -0,0 +1,15 @@
+//
+
+.global-footer {
+  @include global-footer;
```
I've put too much in to the mixin and should tease out the elements up and into this level.

sherakama commented on core/scss/utilities/mixins/global-footer/index.scss:
```diff
@@ -0,0 +1,112 @@
+@charset "UTF-8";
```
This mixin has too much in it. Reduce and move into the component scss file.

sherakama commented on core/scss/utilities/mixins/global-footer/index.scss:
```diff
@@ -0,0 +1,112 @@
+  .global-footer__container {
+    @include display-grid;
+
+    @media #{$breakpoint-sm} {
```
TODO: Create/Find a mixin for these. Seems too much effort to write all this.

sherakama commented on core/scss/variables/core/_breakpoints.scss:
```diff
@@ -242,60 +242,3 @@ $breakpoint-lg-only: 'only screen and (min-width: #{$screen-lg}) and (max-width:
-////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
-
-//
-// $grid-xs
```
None of these were used anywhere so I am assuming these are leftovers from the neat days.

josephgknox replied:
Indeed, they were.

sherakama commented on core/templates/components/atoms/media-block-body/media-block-body.twig:
```diff
@@ -1,4 +1,4 @@
-<div{{ attributes }} class="decanter-media-block__body {{ classes }}">
+<div{{ attributes }} class="decanter-media-block__body {{ modifier_class }}">
```
The convention is to provide a `modifier_class` variable so others can inject their modifiers.

sherakama commented on core/templates/components/molecules/icon-block/icon-block.twig:
```diff
@@ -1,4 +1,4 @@
-<div class="decanter-icon-block {{ classes }}">
+<div class="decanter-icon-block {{ modifier_class }}">
```
This block is missing its attributes twig variable.

sherakama commented on core/templates/components/atoms/media-block-img/media-block-img.twig:
```diff
@@ -1 +1 @@
-<img class="decanter-media-block__img {{ classes }}" src="{{ src|default("../../../img/circle.png") }}" alt="{{ alt|default("Please provide alt text") }}">
+<img class="decanter-media-block__img {{ modifier_class }}" src="{{ src|default("../../../img/circle.png") }}" alt="{{ alt|default("Please provide alt text") }}">
```
This block is missing its attributes twig variable.

sherakama commented on core/templates/components/molecules/icon-grid/icon-grid.twig:
```diff
@@ -1,12 +1,12 @@
   {% set classes = "" %}
     {% for item in items %}
-      {% include "../../molecules/media-block-row/media-block-row.twig" with item %}
+      {% include "@decanter/components/molecules/media-block-row/media-block-row.twig" with item %}
```
Should always use the namespace.

sherakama commented on core/templates/components/organisms/accordions/accordions.twig:
```diff
@@ -1,4 +1,4 @@
-<ul class="decanter-accordion {{ classes }} {{ modifier_class }}">
+<ul class="decanter-accordion {{ modifier_class }} {{ modifier_class }}">
```
This block is missing its attributes twig variable.
Double modifier_class option.

sherakama commented on core/templates/components/organisms/global-footer/global-footer.twig:
```diff
@@ -0,0 +1,31 @@
+      <nav class="global-footer__nav" role="navigation">
+        <ul class="global-footer__menu global-footer__menu--global">
+          <li><a href="https://www.stanford.edu">Stanford Home</a></li>
+          <li><a href="https://visit.stanford.edu/plan/">Maps & Directions</a></li>
```
unencoded ampersand.

sherakama commented on kss/kss-config.json:
```diff
@@ -9,7 +9,7 @@
   "builder": "builder/decanter",
   "extend-drupal8": true,
-  "namespace": "decanter:templates",
+  "namespace": "decanter:core/templates",
```
New path due to new directory structure just merged in.

sherakama commented on scss/kss/_index.scss:
```diff
@@ -1,15 +0,0 @@
-@charset "UTF-8";
```
This was removed and moved into the custom theme.

sherakama commented on core/scss/utilities/mixins/global-footer/index.scss:
```diff
@@ -0,0 +1,112 @@
+    font-weight: 300;
+  }
+
+  .global-footer__menu {
```
Some of these styles could be better refactored into a mixin for a vertical menu.

sherakama commented on core/scss/utilities/mixins/layout/_centered-grid-container.scss:
```diff
@@ -0,0 +1,31 @@
+//
+// Centered Grid Container
+//
+// Creates a responsive centered grid for all breakpoints
```
Needs better documentation. Like, how when using you can only have one child element to what you have applied this to.

sherakama commented on core/scss/utilities/mixins/layout/_grid-media.scss:
```diff
@@ -0,0 +1,94 @@
+//
+// @grid-media($breakpoint: 'xs', $grid: '1fr 1fr 1fr 1fr')
+//
+// A mixin for quickly defining a grid at a provided breakpoint.
```
Needs more documentation and examples.

sherakama commented on core/scss/utilities/mixins/layout/_grid-media.scss:
```diff
@@ -0,0 +1,94 @@
+//
+// Styleguide Mixins.Layout.Grid.Grid Media
+//
+@mixin grid-media($breakpoint: 'xs', $columns: 1fr 1fr 1fr 1fr, $gutters: 1) {
```
Is this a good name for this?

sherakama commented on core/scss/utilities/mixins/layout/_grid-media.scss:
```diff
@@ -0,0 +1,94 @@
+//
+// Styleguide Mixins.Layout.Grid.Grid Media
+//
+@mixin grid-media-only($breakpoint: 'xs', $columns: 1fr 1fr 1fr 1fr, $gutters: 1) {
```
This should be separated out in to its own file as the 1-file-per-mixin rule should be maintained.

josephgknox commented on core/scss/components/organisms/_global-footer.scss:
```diff
@@ -0,0 +1,141 @@
+  }
+
+  .global-footer__menu--policy {
+    font-size: 1.5rem;
```
Believe this is a little smaller than homesite.

josephgknox commented on core/scss/components/organisms/_global-footer.scss:
```diff
@@ -0,0 +1,141 @@
+  }
+
+  .global-footer__copyright {
+    font-size: 1.5rem;
```
Believe this is a little smaller than homesite.

josephgknox commented on core/scss/components/organisms/_global-footer.scss:
```diff
@@ -0,0 +1,141 @@
+        font-size: 3.3rem;
+      }
+
+      font-size: 3rem;
```
Believe this is a little smaller than homesite.

josephgknox commented on core/scss/utilities/mixins/icons-logos/_ligature-logo.scss:
```diff
@@ -0,0 +1,28 @@
+@charset 'UTF-8';
+
+//
+// @ligature-logo
```
This is cool!

josephgknox commented on core/scss/components/organisms/_global-footer.scss:
```diff
@@ -0,0 +1,141 @@
+  @include centered-grid-container;
+  background-color: $color-cardinal-red;
+  color: $color-white;
+  padding: 2.5rem 0;
```
Effort is the same in writing, so I am open to arguments against, but I am a fan of using mixins where and when we can. This could be `@include padding(2.5rem 0);`

josephgknox commented on core/scss/components/organisms/_global-footer.scss:
```diff
@@ -0,0 +1,141 @@
+  }
+
+  .global-footer__brand {
+    margin: 0 0 1.25rem;
```
Effort is the same in writing, so I am open to arguments against, but I am a fan of using mixins where and when we can. This could be @include margin(0 0 1.25rem);

josephgknox commented on core/scss/components/organisms/_global-footer.scss:
```diff
@@ -0,0 +1,141 @@
+      justify-items: center;
+    }
+
+    @include grid-media('sm', 15rem auto) {
```
I like these.

sherakama commented on core/templates/components/molecules/icon-grid/icon-grid.twig:
```diff
@@ -1,12 +1,12 @@
-<div{{ attributes }} class="decanter-icon-grid {{ classes }}">
+<div{{ attributes }} class="decanter-icon-grid {{ modifier_class }}">
```
Thanks for adding the modifier_class.

sherakama commented on core/templates/components/molecules/media-block-row/media-block-row.twig:
```diff
@@ -1,12 +1,12 @@
-<div{{ attributes }} class="decanter-icon-grid__row {{ classes }}">
+<div{{ attributes }} class="decanter-icon-grid__row {{ modifier_class }}">
```
This one too.

sherakama commented:
@josephgknox,

I have cleaned up the CodeClimate issues other than an incorrect check. I have also portioned out the mixins to their own files and updated the documentation on them.

Please have a review and I think it is good to go.

josephgknox commented:
Awesome! Thanks, Shea!
